code/workspaces/infrastructure-api/src/auth/permissionMiddleware.js

```diff
@@ -3,25 +3,27 @@ import logger from '../config/logger';
 
 const permissionDelim = ':';
 const projectName = 'project';
+const systemName = 'system';
 
 function permissionWrapper(permissionSuffix) {
-  const requiredPermission = projectName.concat(permissionDelim, permissionSuffix);
+  const requiredProjectPermission = projectName.concat(permissionDelim, permissionSuffix);
+  const requiredSystemPermissions = systemName.concat(permissionDelim, permissionSuffix);
 
   return (request, response, next) => {
     const grantedPermissions = get(request, 'user.permissions') || [];
 
     logger.debug('Auth: checking permissions');
     logger.debug(`Auth: expected permission suffix: ${permissionSuffix}`);
-    logger.debug(`Auth: expected permission: ${requiredPermission}`);
+    logger.debug(`Auth: expected permission: ${requiredProjectPermission} || ${requiredSystemPermissions}`);
     logger.debug(`Auth: granted user permissions: ${grantedPermissions}`);
 
-    if (grantedPermissions.includes(requiredPermission)) {
+    if (grantedPermissions.includes(requiredProjectPermission) || grantedPermissions.includes(requiredSystemPermissions)) {
```

Comment:
I don't think this needs to be this clever. I think you can just change the second half to
```
grantedPermissions.includes(SYSTEM_ADMIN)
```
where `SYSTEM_ADMIN = 'system:instance:admin'`

Reply:
Again, I think it would be helpful if we consistently used projectKey rather than projectName.

Reply:
This makes sense based on the earlier commend saying that instance admin is all powerful. I will update the check to the way you have suggested.

Agreed that it would be good to start trying to get some constancy between use of `projectName` and `projectKey`. I'll update this file to use `projectKey`.